moq-web/src/session.rs

```diff
@@ -3,23 +3,22 @@ use url::Url;
 
 use crate::{Error, Result};
 
-pub async fn connect(addr: &Url) -> Result<moq_transfork::Session> {
+pub async fn connect(addr: &mut Url) -> Result<moq_transfork::Session> {
```

Comment:
IMO Don't make this `&mut`. Instead either:
1. Clone the `&Url` if needed (http only).
2. Take ownership of the `Url` (caller will need to `clone()`).

The current code destructively changes the protocol of `self.src` which might cause a bug if we ever try to reconnect.

Reply:
I have to transform this URL  to https so that it works with WebTransport. I guess I can do that in the place where you create the MoQ connection

Reply:
Instead of:

```rust
let _ = addr.set_scheme("https");
```

You can do:
```rust
// Make a copy of the address, changing it from HTTP to HTTPS for WebTransport:
let mut addr = addr.clone();
let _ = addr.set_scheme("https");
```

Reply:
Oh yeah that makes sense. I didn't realize the actual MoQ session gets created in the connect function lol